### PR TITLE
feat: Support pod spec for label and cronjobs

### PR DESCRIPTION
**Pull request by msvticket**
This PR adds support for:

- Setting labels in pod spec
- Setting labels and annotations in pod spec of cronjobs
- Removing annotations and labels, including from pod specs
- *Not* overriding existing annotation / label.

Tests for new features and some old are added. I haven't bothered adding tests for all features for both annotations and labels since the same code is now used for these commands.
Code duplication is reduced by making `label` and `annotate` use the same code.

A lot of linting errors after upgrade of linter are fixed.

I committed the result of `make fmt`.

### CODE REVIEW

File: pkg/tagging/base.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	cmd.Flags().StringVarP(&o.Dir, "dir", "", ".", "the directory to recursively look for the *.yaml or *.yml files")
+	cmd.Flags().BoolVarP(&o.PodSpec, "pod-spec", "p", false,
+		fmt.Sprintf("%s the PodSpec in spec.template.metadata.%ss (or spec.jobTemplate.spec.template.metadata.%[2]ss for CronJobs) rather than the top level %[2]ss", tagVerb, tagType))
+	cmd.Flags().BoolVar(&o.Overwrite, "overwrite", true, "Set to false to not overwrite any existing value")
+	o.Filter.AddFlags(cmd)
+	return cmd, o
+}
+
+// UpdateTagInYamlFiles updates the annotations in yaml files
+func UpdateTagInYamlFiles(dir string, tagType string, tags []string, filter kyamls.Filter, podSpec bool, override bool) error { //nolint:gocritic
```
Comment (ankitm123) [inline pkg/tagging/base.go:66]: Will this function be used somewhere else other than this package? If not, we dont need to export this

Comment (msvticket) [reply pkg/tagging/base.go:66]: It is used in other places in this PR

File: docs/cmd/jx-gitops_annotate.md [hunk 2]
```diff
@@ -15,9 +15,11 @@ Annotates all kubernetes resources in the given directory tree
 ### Examples
 
   # updates recursively annotates all resources in the current directory
-  jx-gitops annotate myannotate=cheese another=thing
+  jx-gitops annotate myannotation=cheese another=thing
   # updates recursively all resources
   jx-gitops annotate --dir myresource-dir foo=bar
+  # remove annotations
+  jx-gitops annotate myannotate- another-
```
Comment (tomhobson) [inline docs/cmd/jx-gitops_annotate.md:22]: is this a standard way to annotate?
Seems quite strange

File: docs/man/man1/jx-gitops-label.1 [hunk 3]
```diff
@@ -5,7 +5,7 @@
```
Comment (tomhobson) [inline docs/man/man1/jx-gitops-label.1:5]: man1/*.1?



Is the file name correct here?

File: docs/man/man1/jx-gitops-annotate.1 [hunk 4]
```diff
@@ -41,9 +41,13 @@ Annotates all kubernetes resources in the given directory tree
     adds Kubernetes resource kinds to exclude. For kind expressions see:
```
Comment (tomhobson) [inline docs/man/man1/jx-gitops-annotate.1:41]: man1/*.1?



Is the file name correct here?

File: pkg/tagging/base.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+					&yaml.Node{
+						Kind:  yaml.ScalarNode,
+						Value: k,
+					},
+					vn.YNode())
+				modified = true
+			}
+		}
+		return modified, nil
+	}
```
Comment (tomhobson) [inline pkg/tagging/base.go:124]: this is suuuper high cognitive complexity, we should tyr to split it up a little I think so it's more understandable

File: pkg/tagging/base.go (hunk 1 above)
Comment (tomhobson) [inline pkg/tagging/base.go:66]: `//nolint:gocritic`
We shouldn't do this for actual production code, should only use it for generated code and the other IMHO

**Review by tomhobson — CHANGES_REQUESTED** [review]

File: docs/cmd/jx-gitops_annotate.md (hunk 2 above)
Comment (msvticket) [reply docs/cmd/jx-gitops_annotate.md:22]: Yes, it's how you remove an annotation with `kubectl annotate`

File: docs/man/man1/jx-gitops-annotate.1 (hunk 4 above)
Comment (msvticket) [reply docs/man/man1/jx-gitops-annotate.1:41]: Yes. Look how man files are named in your system.

File: pkg/tagging/base.go (hunk 1 above)
Comment (msvticket) [reply pkg/tagging/base.go:66]: Well, I haven't really thought about. It was there when I started... But I'll remove it.

File: pkg/tagging/base.go (hunk 5 above)
Comment (msvticket) [reply pkg/tagging/base.go:124]: Well, that's a matter of opinion... But I'll split it up a bit

**Review by tomhobson — APPROVED** [review]
Looks way better

**Pull request merged**